### PR TITLE
Avoid to read same accepted blocks twice from disk

### PR DESCRIPTION
**Pull request by iFrostizz**
Closes https://github.com/ava-labs/hypersdk/issues/1165

### CODE REVIEW

**Review by aaronbuchwald — CHANGES_REQUESTED** [review]
Could we just use:

```golang
		blkID, err := vm.GetBlockIDAtHeight(ctx, height)
		if err != nil {
			return fmt.Errorf("could not find accepted block at height %d: %w", height, err)
		}

		blk, err := vm.GetStatelessBlock(ctx, blkID)
		if err != nil {
			return fmt.Errorf("could not find accepted block (%s) at height %d: %w", blkID, height, err)
		}
```

inside of the for loop so we re-use the existing functions that already check the cache

**Comment by iFrostizz** [comment]
Thanks @aaronbuchwald , updated the `GetCachedBlock` function so that it uses these already-written functions. Placed them inside of the `GetCachedBlock` function to avoid repetition.

**Comment by aaronbuchwald** [comment]
> Thanks @aaronbuchwald , updated the `GetCachedBlock` function so that it uses these already-written functions. Placed them inside of the `GetCachedBlock` function to avoid repetition.

I think we can just get rid of the new function completely and use https://github.com/ava-labs/hypersdk/pull/1214#pullrequestreview-2206378640 in the for loop body

File: vm/vm.go [hunk 1]
```diff
@@ -1171,7 +1171,12 @@ func (vm *VM) loadAcceptedBlocks(ctx context.Context) error {
 		start = vm.lastAccepted.Hght - lookback
 	}
 	for i := start; i <= vm.lastAccepted.Hght; i++ {
-		blk, err := vm.GetDiskBlock(ctx, i)
+		blkID, err := vm.GetBlockIDAtHeight(ctx, i)
+		if err != nil {
+			vm.snowCtx.Log.Info("could not find block ID on-disk", zap.Uint64("height", i))
+			continue
+		}
+		blk, err := vm.GetStatelessBlock(ctx, blkID)
```
Comment (aaronbuchwald) [inline vm/vm.go:1179]: This code should be unchanged, this is called on startup when there are no blocks in the cache, so we need to load all of these from disk and can call `GetDiskBlock` directly

Comment (iFrostizz) [reply vm/vm.go]: Thanks, done!

**Review by aaronbuchwald — APPROVED** [review]

**Pull request merged**